chapter07/chapter07.adoc

```diff
@@ -0,0 +1,276 @@
+        return productService.getProduct(productId);
+    }
+
+    // …
+----
+  
+It will expose a metric called `productLookupTime`, which will track the amount of time spent in the `getProduct()` method in milliseconds.
+You can visit the following URL `++https://localhost:<port>/metrics?scope=application++` (Replace `<port>` with the actual port where the server is running) to see the Response time of this method as below:
+
+[source]
+----
+…
+# HELP productLookupTime_seconds_max Time needed to lookup for a products
+# TYPE productLookupTime_seconds_max gauge
+productLookupTime_seconds_max{method="getProduct",mp_scope="application",} 0.002270643
+…
+----
+
+==== Tracking number of invocations
+MicroProfile Metrics also allows you to track the number of invocations of a method as a counter metric. The code example below shows how to use the `@Counted` annotation to track the invocation count.
+
+[source, java]
+----
+import org.eclipse.microprofile.metrics.Metrics;
+import org.eclipse.microprofile.metrics.MetricType;
+
+public class ProductResource {
+
+    // Expose the invocation count as a counter metric
+    @Counted(name = "productAccessCount",
+        absolute = true,
+        description = "Number of times the list of products is requested")
+    public Response getProducts() {
+       // Method implementation
+       // ....
+    }
+}
+----
+  
+In the example above, the `@Counted` annotation tells MicroProfile Metrics to track the number of invocations of the `getProducts()` method and expose this metric as a counter. The name, and description of the metric can also be specified. 
+You can visit the following URL `++https://localhost:<port>/metrics?scope=application++` (Replace `<port>` with the actual port where the server is running) to see the number of times this method is called as below:
+
+[source]
+----
+…
+# HELP productAccessCount_total Number of times the list of products is requested
+# TYPE productAccessCount_total counter
+productAccessCount_total{mp_scope="application",} 3.0
+…
+----
+
+=== Creating a Custom Metric
+
+Creating a custom metric to track the number of products in a catalog involves using the MicroProfile Metrics API. This custom metric can be implemented as a gauge, which measures an instantaneous value (in this case, the current number of products in the catalog). 
+
+[source, java]
+----
+import org.eclipse.microprofile.metrics.annotation.Gauge;
+… 
+
+@Path("/products")
+@ApplicationScoped
+public class ProductResource {
+  // … 
+   
+  @GET
+  @Path("/count")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Gauge(name = "productCatalogSize", 
+    unit = "none", 
+  description = "Current number of products in the catalog")
+  public long getProductCount() {
+     return productCatalogSize;
+  }
+}
+----
+
+The gauge metric `productCatalogSize` can be accessed through the following endpoint: 
+
+`/metrics?name=io_microprofile_tutorial_store_product_resource_ProductResource_productCatalogSize`
```

Comment:
as the doc doesn't show the package name, it would probably be better to set absolute=true and just use the absolute name here.

Reply:
Vendors may, by their on implementation, support `/metrics?name=<name>` to directly retrieve that metric from all scopes. But the specification itself only illustrates `/metrics?scope=<scope>&name=<name>`

Reply:
Fixed with 4239e8c72e7b156e8433bfd440f8c5823707f76a